server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -778,36 +801,33 @@ private class PipelineIterator implements Iterator<PipelineSlot> {
 
         private final String defaultPipeline;
         private final String finalPipeline;
+        private final List<Pipeline> pluginsPipelines;
+
         private final Iterator<PipelineSlot> pipelineSlotIterator;
 
-        private PipelineIterator(String defaultPipeline, String finalPipeline) {
+        private PipelineIterator(String defaultPipeline, String finalPipeline, List<Pipeline> pluginsPipelines) {
```

Comment:
Is there any consequence to passing the plugin pipeline ID to `PipelineIterator` and deferring plugin pipeline lookup to when the iterator is used?

Reply:
You're right, it's more in line with the current constructor. Changed in 7ce9f5cd1f455b40905b40d676e0c9abbf36c03c 👍